### DISCUSSION TITLE
Add other chains after the initial configureChains has been set

### DISCUSSION THREAD

**Discussion by qqqllppp** [wevm/wagmi#1454] [Ideas]
Is there a way to change the supported chains after initial config has been set? For example:

1. config set during app start-up
```
const { chains, provider, webSocketProvider } = configureChains(
  [polygon],
  [publicProvider()],
)
```

2. some where in a web page of the app
```
addSupportedChains([polygonMumbai])
```

this effectively "adds" a supported chain to the `configureChains`

This would be useful for apps that only want their users to interact with mainnet chains such as Ethereum and Polygon mainnets, but when developer wants to do some testing on the production app, they can click a secret access button to make **only their instance** of the app support testnets.

So normal users would only be able to access: Polygon mainnet
Developers would be able to access: Polygon mainnet and Mumbai testnet

Why is this better than setting both mainnet and testnet in the initial `configureChains`? Because even if we hide the testnet from UI access to normal users, they are still able to access the testnet via switching to the testnet from the wallet (please correct me if this is not true).

**Comment by jxom** [comment]
Adding `configureChains` & `createClient` inside a `useMemo` within the React app might work – where the chains are stored in state which could be modified via React Context.

```tsx
const ChainsContext = createContext(...)

function App() {
  const [supportedChains, setSupportedChains] = useState(...)
  const { chains, provider } = useMemo(() => configureChains(supportedChains, ...), [supportedChains])
  const client = useMemo(() => createClient(...), [])
  
  return (
    <WagmiConfig client={client>
      <ChainsContext.Provider value={setCurrentChains}>
        ...
      </ChainsContext.Provider>
    </WagmiConfig>
  )
}

function Example() {
  const setCurrentChains = useContext(ChainsContext)
  
  ...
    setCurrentChains(x => [...x, polygonMumbai])
  ...
}
```

**Reply by qqqllppp** [reply]
It works, but one issue here is that it does not persists once the page is refreshed
